utils.go

```diff
@@ -0,0 +1,18 @@
+package nmap
+
+// https://siongui.github.io/2018/04/14/go-remove-duplicates-from-slice-or-array/
+func RemoveDuplicatesFromSlice(s []string) []string {
```

Comment:
WDYT of renaming this function to make it explicit that it is only for string slices, and to add a comment to the form `// RemoveDuplicatesFromStringSlice ...`?

Also, if this stays within the `nmap` package it needs to be a private function, not a public one 🤔 If you feel like making a subpackage such as `internal/slices/unique_string.go` containing this func, that would probably be preferable, because:

* Putting it in `internal` ensures that people can't import it in their projects by mistake
* It makes it obvious from the file structure what its purpose is
* It is easily expandable

What do you think? :)